src/main/typescript/wcardinal/ui/d-slider-vertical.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2019 Toshiba Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Point } from "pixi.js";
+import { DSlider, DSliderOptions, DThemeSlider } from "./d-slider";
+import { DSliderTrack } from "./d-slider-track";
+import { DSliderTrackVertical } from "./d-slider-track-vertical";
+
+export interface DSliderVerticalOptions<
+	THEME extends DThemeSliderVertical = DThemeSliderVertical
+> extends DSliderOptions<THEME> {
+
+}
+
+export interface DThemeSliderVertical extends DThemeSlider {
+
+}
+
+// this is space beetween slider value and slider button
+const VERTICAL_PIXEL_BALANCE = 2;
+
+export class DSliderVertical<
+	THEME extends DThemeSliderVertical = DThemeSliderVertical,
+	OPTIONS extends DSliderVerticalOptions<THEME> = DSliderVerticalOptions<THEME>
+> extends DSlider<THEME, OPTIONS> {
+
+	protected newTrack( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackVertical( options && options.track );
+	}
+
+	protected newTrackSelected( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackVertical( options && options.track );
+	}
+
+	protected assertSize(): void {
+		this._track.setHeight( this.height - this._value.height - VERTICAL_PIXEL_BALANCE );
+		this._trackSelected.setHeight( this._track.height - VERTICAL_PIXEL_BALANCE );
```

Comment:
The `trackSelected`'s height is updated at the methods `assertSize`, `updateCoordinates` and `updateChildren`. Are you sure that they are all required?